sem/manager.py

```diff
@@ -290,7 +292,32 @@ def check_and_fill_parameters(self, param_list, needs_rngrun):
     # Simulation running #
     ######################
 
-    def run_simulations(self, param_list, show_progress=True, stop_on_errors=True):
+    def parse_log_component(self,log_component):
+        #Verifies if the log levels passed in the log_component dictionary are correct 
+        # and removes the level_ prefix (if present) from the log levels and converts 
+        # the levels to corresponding levels without the level_prefix
+        # For example,
+        # 'level_debug' get converted to 'warn|error|debug'
+        ret_dict = {}
+        log_level_list = ['error','warn','debug','info','function','logic','all']
+        for component,levels in log_component.items():
+            log_levels_comp = set()
+            for level in levels.split('|'):
+                match = re.match(r"(?=("+'|'.join([r'^' + lvl + r'$' for lvl in log_level_list])+r"))",level)
+                lvl_match = re.match(r"(?=("+'|'.join([r'^level_' + lvl + r'$'for lvl in log_level_list])+r"))",level)
```

Comment:
hard to read. could you break this down into multiple lines or even create a separate function if too complex? also comments are welcomed, think about code maintenance in the future

Reply:
on a second thought: regexp might be overkill for the job. you're expecting `level` to be one of a few possible entries. you could hard code these entries in a dictionary, e.g.
```python
converter = {'error': 'error',
             'warn': 'warn',
             # etc
             'level_warn': 'error|warn',
             # etc
}
```
you might also want to check for possible user errors, e.g., how is the case 'warn|level_info' handled? does 'warn' appear twice? are 'warn|info' and 'info|warn' handled in the same way by the new code?

Reply:
what about prefixes? how are they handled?

Reply:
I agree with @mattia-lecci's suggestion to use a converter dictionary.

There is also some confusion in the terminology: in [the documentation](https://www.nsnam.org/docs/manual/html/logging.html), they talk about logging _classes_ to identify `warn`, `function`, etc; instead, logging _levels_ are defined as the class and all above.

By the way, are we sure we want to support logging classes? Just supporting levels would be significantly easier. Then again, I usually rely on levels instead of classes, but other users might have different needs...

Reply:
> you might also want to check for possible user errors, e.g., how is the case 'warn|level_info' handled? does 'warn' appear twice? 

No 'warn' will not appear twice(a set is used for storing the log classes). Every log class will appear only once. 

> are 'warn|info' and 'info|warn' handled in the same way by the new code?

The order of log_levels will not make a difference.


> what about prefixes? how are they handled?

I can add a check for this. If the user tries to add a prefix, we can just drop it.

Reply:
We can use a converter dictionary as well. It might be less prone to errors as well. 

> By the way, are we sure we want to support logging classes? Just supporting levels would be significantly easier. Then again, I usually rely on levels instead of classes, but other users might have different needs...

Supporting log classes and log levels might not be much hassle. It would be better if I code this and then you both can give further comments on this.

Reply:
i agree, adding support for both classes and level is not a big deal if done with a dictionary so we should do it (i mostly use log levels as well.. it would be annoying if they were not supported).


> No 'warn' will not appear twice(a set is used for storing the log classes). Every log class will appear only once.
Good job, I missed that! 

> The order of log_levels will not make a difference.
Are you sure that this won't make a difference? How do you check if the same simulation already exists with the given log components?

> I can add a check for this. If the user tries to add a prefix, we can just drop it.
Yes, this should be fine, you're adding a `prefix_all` anyway